### ISSUE TITLE
All features are reloaded on zoom level change

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#5568]
All FeatureType objects seem to be completely discarded and re-loaded (incl. geometry) on every zoom level change.

Ideally feature objects shared between zoom levels should be preserved/re-used and only geometry is re-loaded if necessary.

**Comment by vng** [comment]
Do you mean by "objects" - C++ FeatureType class instances?
Well, FeatureType is a class that is not intended to be persistent. And it is lightweight (except geometry), so adding any kind of cache may add more overhead than profit.

Better to cache more _high-level_ objects. Drape caches tiles geometry. Search caches cities, villages, etc. Routing caches road's geometry.

**Comment by pastk** [comment]
> Do you mean by "objects" - C++ FeatureType class instances? Well, FeatureType is a class that is not intended to be persistent. And it is lightweight (except geometry), so adding any kind of cache may add more overhead than profit.

Yes, the issue is mostly about **re-reading** feature headers, geometry and meta-data.
I think a standalone cache per se is not necessary indeed.
But just a more smart way of keeping the objects instead of re-loading everything.
I.e. when zoom level is changed and the visibility index is read to form a new list of features - we can compare it to the current list and re-use many objects from there.
